src/arm/mach/init.c

```diff
@@ -54,6 +60,54 @@ struct cpuinfo_arm_isa cpuinfo_isa = {
 #endif
 };
 
+struct cache_array {
+	struct cpuinfo_cache *caches;
+	uint32_t count;
+};
+
+/*
+ * iOS 15 and macOS Monterey 12 added sysctls to describe configuration information
+ * where not all cores are the same (number of cores, cache sizes).
+ * 
+ * Each perflevel sysctl has a prefix of `hw.perflevel??.` where ?? is the
```

Comment:
Are perflevels reported by OS ordered? cpuinfo orders cores and logical processors from the most performant to the least performance, so `cpuinfo_get_core(0)` is the the fastest core and `cpuinfo_get_core(cpuinfo_get_cores_count() - 1)` is the most power-efficient core.